src/duplo_resource/service.py

```diff
@@ -53,7 +53,34 @@ def update_image(self,
     }
     self.duplo.post(f"subscriptions/{tenant_id}/ReplicationControllerChange", data)
     return {"message": f"Successfully updated image for service '{name}'"}
-  
+ 
+  @Command()
+  def bulk_update_image(self, 
+                  serviceimage: args.SERVICEIMAGE):
+    """Update multiple services.
+    
+    Args:
+      serviceimage/-S (string): takes n sets of two arguments, service name and image name. e.g -S service1 image1:tag -S service2 image2:tag
+    """
+    tenant_id = self.tenant["TenantId"]
+    # Check that all services exist before updating
+    servicelist = [item[0] for item in serviceimage]
+    for name in servicelist:
```

Comment:
That's basically throw-away code that grabs the first element from each list in the list of lists (which should be the service).  Then it's verifying all the services exist by doing a find before we go in and actually do the updates.  Reason being if service 3 doesn't exist, you could potentially update service 1 and 2 before 3 fails and have an inconsistent state.   The find function is doing error checking so this returns the error if the service is not found and it bails before attempting any image updates in the main loop.  I could wrap it in a try/except for clarity or something; I'm open to any suggestions you have because I agree that it's a bit redundant.

Reply:
I see the very obviously thing you're pointing out, fixed it.